### ISSUE TITLE
[dev] job stuck in "runnable" state while queu empty

### ISSUE THREAD

**Issue by chrisgorgo** [OpenNeuroOrg/openneuro#120]
https://openneuro.dev.sqm.io/datasets/ds000105/versions/00001?app=mindboggle&version=1

Job has been in "runnnable" state for about an hour now.

**Comment by nellh** [comment]
This job is stuck because it's running into the permission issue that was fixed but is not included in the 0.0.3 tag.

**Comment by chrisgorgo** [comment]
I see - I just triggered a new release of the mindboggle BIDS App with the fix.

I would expect this to fail when running rather than being stuck in runnable. Is there a blacklist of apps/versions with known issues preventing them to run?

**Comment by nellh** [comment]
Yeah, this should hang in starting/running or fail and you are right, this is another issue with this job. I've reopened the Amazon support ticket (4167759301) for this Batch deadlock.

**Comment by chrisgorgo** [comment]
FYI - I deployed 0.0.4 and it still seems to be stuck in RUNNABLE.

**Comment by nellh** [comment]
As a workaround for this, you can test with +/- 500MB of memory from the 8000MB value used. This is caused by ECS starting workers that cannot run the job due to a lack of memory, which is a unit conversion bug in Batch or ECS. The job requests 8388608000 bytes and ECS starts an instance with 8000000000 bytes, which never accepts the job. Upping the memory requested will start a larger instance or lowering it will run on the existing instance.

**Comment by chrisgorgo** [comment]
bumped it to 8500MB - still only RUNNABLE

**Comment by nellh** [comment]
Amazon has acknowledged this bug and escalated it to the Batch team.

It looks like your 8500MB job did run, it was waiting on ECS instances to get allocated and started.

**Comment by chrisgorgo** [comment]
Thanks - they run correctly. Closing.
